batcher/src/types/mod.rs

```diff
@@ -15,22 +16,91 @@ pub enum ProvingSystemId {
 pub struct VerificationData {
     pub proving_system: ProvingSystemId,
     pub proof: Vec<u8>,
-    pub public_input: Option<Vec<u8>>,
+    pub pub_input: Option<Vec<u8>>,
     pub verification_key: Option<Vec<u8>>,
     pub vm_program_code: Option<Vec<u8>>,
+    pub proof_generator_addr: Address,
 }
 
-#[derive(Debug, Default, Serialize, Deserialize)]
-pub struct VerificationBatch(Vec<VerificationData>);
+#[derive(Debug, Default)]
+pub struct VerificationDataCommitment {
+    pub proof_commitment: [u8; 32],
+    pub pub_input_commitment: [u8; 32],
+    // This could be either the VM code (ELF, bytecode) or the verification key
+    // depending on the proving system.
+    pub proving_system_aux_data_commitment: [u8; 32],
+    pub proof_generator_addr: [u8; 20],
+}
+
+impl From<&VerificationData> for VerificationDataCommitment {
+    fn from(verification_data: &VerificationData) -> Self {
+        let mut hasher = Keccak256::new();
+
+        // compute proof commitment
+        hasher.update(verification_data.proof.as_slice());
+        let proof_commitment = hasher.finalize_reset().into();
+
+        // compute public input commitment
+        let mut pub_input_commitment = [0u8; 32];
+        if let Some(pub_input) = &verification_data.pub_input {
+            hasher.update(pub_input);
+            pub_input_commitment = hasher.finalize_reset().into();
+        }
+
+        // compute proving system auxiliary data commitment
+        let mut proving_system_aux_data_commitment = [0u8; 32];
+        // FIXME(marian): This should probably be reworked, for the moment when the proving
+        // system is SP1, `proving_system_aux_data` stands for the compiled ELF, while in the case
+        // of Groth16 and PLONK, stands for the verification key.
+        if let Some(vm_program_code) = &verification_data.vm_program_code {
+            debug_assert_eq!(verification_data.proving_system, ProvingSystemId::SP1);
```

Comment:
Debug assert should be removed

Reply:
I agree, I added it just as an extra check to make the development process easier. That's why I left the fixme but it could be removed